### PR TITLE
PMM-5819 added STT tests

### PR DESCRIPTION
**Pull request by yurkovychv**

### CODE REVIEW

File: server/settings_test.go [hunk 1]
```diff
@@ -72,6 +73,188 @@ func TestSettings(t *testing.T) {
...
+							DisableTelemetry: true,
+						},
+						Context: pmmapitests.Context,
+					})
+					pmmapitests.AssertAPIErrorf(t, err, 400, codes.InvalidArgument, `cannot disable telemetry while STT is enabled`)
+					assert.Empty(t, res)
+				})
+			})
+
+			t.Run("DisableSTTWhileItIsDisabled", func(t *testing.T) {
```
Comment (BupycHuk) [inline server/settings_test.go:244]: looks like duplicated test

File: server/settings_test.go [hunk 2]
```diff
@@ -72,6 +73,188 @@ func TestSettings(t *testing.T) {
...
+
+				resg, err := serverClient.Default.Server.GetSettings(&server.GetSettingsParams{
+					Body:    server.GetSettingsDefaultBody{},
+					Context: pmmapitests.Context,
+				})
+				require.NoError(t, err)
+				assert.True(t, resg.Payload.Settings.TelemetryEnabled)
+
+				// Restore the STT state to false
+				res, err = serverClient.Default.Server.ChangeSettings(&server.ChangeSettingsParams{
```
Comment (BupycHuk) [inline server/settings_test.go:184]: we have teardown for that

File: server/settings_test.go [hunk 3]
```diff
@@ -72,6 +73,188 @@ func TestSettings(t *testing.T) {
...
+						EnableStt: true,
+					},
+					Context: pmmapitests.Context,
+				})
+				require.NoError(t, err)
+
+				assert.True(t, res.Payload.Settings.SttEnabled)
+				assert.Empty(t, err)
+
+				resg, err := serverClient.Default.Server.GetSettings(&server.GetSettingsParams{
```
Comment (BupycHuk) [inline server/settings_test.go:176]: let's make checks like this (via GetSettings) everywhere we expect that settings should be updated

File: server/settings_test.go (hunk 1 above)
Comment (yurkovychv) [reply server/settings_test.go:244]: we do not have such test before, thats basically the Current state check

File: server/settings_test.go [hunk 4]
```diff
@@ -72,6 +73,188 @@ func TestSettings(t *testing.T) {
...
+				require.NoError(t, err)
+				assert.False(t, res.Payload.Settings.SttEnabled)
+				assert.False(t, res.Payload.Settings.TelemetryEnabled)
+				assert.Empty(t, err)
+			})
+
+			t.Run("EnableSTTWhileTelemetryEnabled", func(t *testing.T) {
+				defer teardown(t)
+
+				// Ensure Telemetry is enabled
```
Comment (yurkovychv) [inline server/settings_test.go:155]: do we need to run this? basically that check is in teardown func

File: server/settings_test.go [hunk 5]
```diff
@@ -72,6 +73,188 @@ func TestSettings(t *testing.T) {
...
+
+				assert.True(t, res.Payload.Settings.SttEnabled)
+				assert.Empty(t, err)
+
+				resg, err := serverClient.Default.Server.GetSettings(&server.GetSettingsParams{
+					Body:    server.GetSettingsDefaultBody{},
+					Context: pmmapitests.Context,
+				})
+				require.NoError(t, err)
+				assert.True(t, resg.Payload.Settings.TelemetryEnabled)
```
[the remark this answers is not in the document]
Comment (yurkovychv) [reply server/settings_test.go:195]: done

File: server/settings_test.go (hunk 1 above)
Comment (BupycHuk) [reply server/settings_test.go:266]: https://github.com/Percona-Lab/pmm-api-tests/pull/91/files#diff-356be822aecac50b5a0a1a1b9efe0676R200

File: server/settings_test.go [hunk 6]
```diff
@@ -41,6 +42,8 @@ func TestSettings(t *testing.T) {
 				bodySettings := res.Payload.Settings
 				res, err := serverClient.Default.Server.ChangeSettings(&server.ChangeSettingsParams{
 					Body: server.ChangeSettingsBody{
+						DisableStt:      !bodySettings.SttEnabled,
```
Comment (BupycHuk) [inline server/settings_test.go:45]: :+1: could you fix it for telemetry too, please

File: server/settings_test.go (hunk 4 above)
Comment (BupycHuk) [reply server/settings_test.go:173]: yes, why not

File: server/settings_test.go (hunk 1 above)
Comment (yurkovychv) [reply server/settings_test.go:266]: thanks, missed that one

File: server/settings_test.go (hunk 6 above)
Comment (yurkovychv) [reply server/settings_test.go]: done

**Comment by BupycHuk** [comment]
If it's ready for review, please click the button and request review from @percona-csalguero

**Comment by AlekSi** [comment]
@yurkovychv there is a merge conflict now

File: server/settings_test.go [hunk 7]
```diff
@@ -72,6 +77,193 @@ func TestSettings(t *testing.T) {
...
+				require.NoError(t, err)
+				assert.False(t, res.Payload.Settings.SttEnabled)
+				assert.Empty(t, err)
+
+				resg, err := serverClient.Default.Server.GetSettings(nil)
+				require.NoError(t, err)
+				assert.True(t, resg.Payload.Settings.TelemetryEnabled)
+				assert.False(t, resg.Payload.Settings.SttEnabled)
+			})
+			t.Run("STTEnabledState", func(t *testing.T) {
```
Comment (BupycHuk) [inline server/settings_test.go:219]: Let's make the same kind of tests with disabled telemetry.

Comment (yurkovychv) [reply server/settings_test.go:219]: Done

**Comment by BupycHuk** [comment]
could you update dependencies via dep ensure

**Comment by BupycHuk** [comment]
probably you forgot to add some file from vendor directory

**Comment by BupycHuk** [comment]
Travis ci tests are failing

**Pull request merged**